### PR TITLE
archive:revert using archive full path name and replenish CLEAN AROBJS

### PR DESCRIPTION
**Pull request by xuxin930**
## Summary
the `P` option is not a POSIX standard and will not perform the replace operation,
this will cause the size of the target static library to become larger.

let's workaround using other implementaion in the Application.mk of apps in https://github.com/apache/nuttx-apps/pull/2138

the new implementation defines `AROJBS`, which is a renamed object file participating in the archive.
so these files need to be deleted in the `CLEAN` macro.

## Impact
Depends on (merge after): https://github.com/apache/nuttx-apps/pull/2138

## Testing
CI Build

### CODE REVIEW

**Comment by acassis** [comment]
@xuxin930 ping

File: tools/Config.mk [hunk 1]
```diff
@@ -589,7 +589,7 @@ define CLEAN
 endef
 else
 define CLEAN
-	$(Q) rm -f *$(OBJEXT) *$(LIBEXT) *~ .*.swp $(OBJS) $(BIN) $(BIN).lock $(EXTRA)
+	$(Q) rm -f *$(OBJEXT) *$(LIBEXT) *~ .*.swp $(AROBJS) $(OBJS) $(BIN) $(BIN).lock $(EXTRA)
```
Comment (anchao) [inline tools/Config.mk:592]: if AROBJS just use for application build, please move this change into rule of application clean

**Comment by xuxin930** [comment]
Duplicate modification with https://github.com/apache/nuttx/pull/11007 
closed

File: tools/Config.mk (hunk 1 above)
Comment (xuxin930) [reply tools/Config.mk:592]: > if AROBJS just use for application build, please move this change into rule of application clean

DONE
in https://github.com/apache/nuttx-apps/pull/2138
closed this PR with duplicate modification https://github.com/apache/nuttx/pull/11007

**Pull request closed without merging**